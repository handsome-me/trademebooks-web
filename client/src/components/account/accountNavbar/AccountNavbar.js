import React from 'react'
import PropTypes from 'prop-types'
import { MDBNav, MDBNavItem, MDBNavLink, MDBIcon } from 'mdbreact'

const AccountNavbar = ({
  toggleJustified,
  activeItemJustified
}) => {
  return (
    <>
      <MDBNav tabs className="border-bottom" color="indigo">
        <MDBNavItem>
          <MDBNavLink
            link
            to="#"
            active={activeItemJustified === '1'}
            onClick={toggleJustified('1')}
            role="tab"
          >
            <MDBIcon icon="user" /> User
          </MDBNavLink>
        </MDBNavItem>
        <MDBNavItem>
          <MDBNavLink
            link
            to="#"
            active={activeItemJustified === '2'}
            onClick={toggleJustified('2')}
            role="tab"
          >
            <MDBIcon icon="store" /> Bookstore
          </MDBNavLink>
        </MDBNavItem>
        <MDBNavItem>
          <MDBNavLink
            link
            to="#"
            active={activeItemJustified === '3'}
            onClick={toggleJustified('3')}
            role="tab"
          >
            <MDBIcon icon="envelope" /> Notifications
          </MDBNavLink>
        </MDBNavItem>
      </MDBNav>
    </>
  )
}

AccountNavbar.propTypes = {
  toggleJustified: PropTypes.func.isRequired,
  activeItemJustified: PropTypes.string.isRequired
}


export default AccountNavbar
