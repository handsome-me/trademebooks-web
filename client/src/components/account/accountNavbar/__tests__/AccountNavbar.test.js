import React from 'react'
import renderer from 'react-test-renderer'
import { MemoryRouter } from 'react-router-dom';
import AccountNavbar from '../AccountNavbar'

describe('Component: AccountNavbar', () => {
  const props = {
    toggleJustified: jest.fn(),
    activeItemJustified: 'user'
  }

  it('should match the snapshot', () => {
    const accountNavbarSnapshot = renderer.create(
      <MemoryRouter>
        <AccountNavbar {...props} />
      </MemoryRouter>
    ).toJSON()
    expect(accountNavbarSnapshot).toMatchSnapshot()
  })
})
