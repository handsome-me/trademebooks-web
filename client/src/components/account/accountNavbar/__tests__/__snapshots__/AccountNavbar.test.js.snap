// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: AccountNavbar should match the snapshot 1`] = `
<ul
  className="nav md-tabs tabs-indigo border-bottom"
  data-test="nav"
>
  <li
    className="nav-item"
    data-test="nav-item"
  >
    <a
      className="nav-link Ripple-parent nav-link Ripple-parent"
      data-test="nav-link"
      href="/"
      onClick={[Function]}
      onMouseUp={[Function]}
      onTouchStart={[Function]}
      role="tab"
    >
      <i
        className="fa fa-user"
        data-test="fa"
      />
       User
      <div
        className="Ripple "
        data-test="waves"
        style={
          Object {
            "height": "0px",
            "left": "0px",
            "top": "0px",
            "width": "0px",
          }
        }
      />
      <div
        className="Ripple "
        data-test="waves"
        style={
          Object {
            "height": "0px",
            "left": "0px",
            "top": "0px",
            "width": "0px",
          }
        }
      />
    </a>
  </li>
  <li
    className="nav-item"
    data-test="nav-item"
  >
    <a
      className="nav-link Ripple-parent nav-link Ripple-parent"
      data-test="nav-link"
      href="/"
      onClick={[Function]}
      onMouseUp={[Function]}
      onTouchStart={[Function]}
      role="tab"
    >
      <i
        className="fa fa-store"
        data-test="fa"
      />
       Bookstore
      <div
        className="Ripple "
        data-test="waves"
        style={
          Object {
            "height": "0px",
            "left": "0px",
            "top": "0px",
            "width": "0px",
          }
        }
      />
      <div
        className="Ripple "
        data-test="waves"
        style={
          Object {
            "height": "0px",
            "left": "0px",
            "top": "0px",
            "width": "0px",
          }
        }
      />
    </a>
  </li>
  <li
    className="nav-item"
    data-test="nav-item"
  >
    <a
      className="nav-link Ripple-parent nav-link Ripple-parent"
      data-test="nav-link"
      href="/"
      onClick={[Function]}
      onMouseUp={[Function]}
      onTouchStart={[Function]}
      role="tab"
    >
      <i
        className="fa fa-envelope"
        data-test="fa"
      />
       Notifications
      <div
        className="Ripple "
        data-test="waves"
        style={
          Object {
            "height": "0px",
            "left": "0px",
            "top": "0px",
            "width": "0px",
          }
        }
      />
      <div
        className="Ripple "
        data-test="waves"
        style={
          Object {
            "height": "0px",
            "left": "0px",
            "top": "0px",
            "width": "0px",
          }
        }
      />
    </a>
  </li>
</ul>
`;
